main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.language.ast.Symbol
+import ca.uwaterloo.flix.language.phase.jvm.JvmName.{Delimiter, DevFlixRuntime, JavaLang, RootPackage}
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+/**
+  * Represents all Flix types that are objects on the JVM (array is an exception).
+  */
+sealed trait BackendObjType {
+  /**
+    * The `JvmName` that represents the type `Ref(Int)` refers to `Ref$Int`.
+    */
+  val jvmName: JvmName = this match {
+    case BackendObjType.Unit => JvmName(DevFlixRuntime, "Unit")
+    case BackendObjType.BigInt => JvmName(List("java", "math"), "BigInteger")
+    case BackendObjType.String => JvmName(JavaLang, "String")
+    case BackendObjType.Channel(_) => JvmName(List("ca", "uwaterloo", "flix", "runtime", "interpreter"), "Channel")
+    case BackendObjType.Lazy(tpe) => JvmName(RootPackage, s"Lazy$Delimiter${tpe.toErased}")
+    case BackendObjType.Ref(tpe) => JvmName(RootPackage, s"Ref$Delimiter${tpe.toErased}")
+    case BackendObjType.Tuple(elms) => JvmName(RootPackage, s"Tuple${elms.length}$Delimiter${erasedListOfTypes(elms)}")
+    case BackendObjType.Enum(_, _) => throw InternalCompilerException("Enum JVM NAME") // TODO
```

Comment:
Are these reachable or not? What happens if I have a reference of a record or any of the other cases that throw exceptions?

Reply:
These are only produced in the new Gen.. code so they are unreachable right now. I will introduce them as they come up. If they appear the compiler blows up

Reply:
But now we are working on the actual compiler backend, so I am not comfortable introducing code that has exceptions. Can you leave them out or comment them for now? (As in leave out the definitions of e.g. BackendObjType.Enum).